collectors/nodestats/nodestats_collector_test.go

```diff
@@ -147,3 +148,62 @@ func TestCollectError(t *testing.T) {
 		t.Error("Expected err not to be nil")
 	}
 }
+
```

Comment:
This is the sort of test that ChatGPT generates! Was curious to try doing so... it's a bit wordy but checks all the conditions!

Reply:
a loooot of code in this repo was generated by ChatGPT. I always follow the same routine, I feed it with the fragment that it needs to test, and show an example of my own not-AI written fragment of test and tell it to follow the same code style. Then I do code review as many times as needed for the code to be enough high quality.